### PR TITLE
Improve string formatting and change method name

### PR DESCRIPTION
**Pull request by dwalluck**

### CODE REVIEW

File: src/main/java/com/redhat/red/build/finder/report/GAVReport.java [hunk 1]
```diff
@@ -16,18 +16,20 @@
...
 public class GAVReport extends Report {
-    private List<String> gavs;
+    private Set<String> gavs;
 
     public GAVReport(List<KojiBuild> builds) {
-        List<KojiBuildInfo> buildInfos = builds.stream().filter(b -> b.getType() != null && b.getType().contains("maven")).map(KojiBuild::getBuildInfo).collect(Collectors.toList());
-        this.gavs = buildInfos.stream().map(b -> b.getMavenGroupId() + ":" + b.getMavenArtifactId() + ":" + b.getMavenVersion()).collect(Collectors.toList());
-        this.gavs.sort(String::compareToIgnoreCase);
+        // FIXME: Will check that !b.getTypes().contains("rpm") instead of getBuildInfo().getMavenArtifactId().endsWith("-repolib"), but requires NOS-1152, NOS-1153
+        List<KojiBuildInfo> buildInfos = builds.stream().filter(b -> b.getTypes() != null && b.getTypes().contains("maven") && !b.getBuildInfo().getMavenArtifactId().endsWith("-repolib")).map(KojiBuild::getBuildInfo).collect(Collectors.toList());
```
Comment (rnc) [inline src/main/java/com/redhat/red/build/finder/report/GAVReport.java]: Do we have a tracker jira to fix this up later?

**Comment by rnc** [comment]
As with #14 should we try to create unit tests for these?

File: src/main/java/com/redhat/red/build/finder/report/GAVReport.java (hunk 1 above)
Comment (dwalluck) [reply src/main/java/com/redhat/red/build/finder/report/GAVReport.java]: I think the workaround is OK as in practice as the only RPMs that have maven artifacts are named '-repolib' by convention. 

This issue has the Kojiji-related issues linked to it. I am not sure if there's a better way to do it.

**Comment by dwalluck** [comment]
> As with #14 should we try to create unit tests for these?

If BuildFinder were refactored to separate out a Main class and make the other methods non-static, then this might make it easier to test functionality.

There is the general problem that unit tests should be offline and avoid any sort of network or database access, so it needs to be decided how to mock the actual Koji API calls like getArchiveTypeMap() which are all methods on a KojiClient object,

**Comment by rnc** [comment]
Agreed with tests not requiring external resource (beyond standard dependencies). I would suggest something like wiremock perhaps?

File: src/main/java/com/redhat/red/build/finder/report/GAVReport.java (hunk 1 above)
Comment (dwalluck) [reply src/main/java/com/redhat/red/build/finder/report/GAVReport.java]: It's the same instance that tracks the tickets for this.

**Comment by dwalluck** [comment]
> Agreed with tests not requiring external resource (beyond standard dependencies). I would suggest something like wiremock perhaps?

Kojiji has a way to load the XML data normally sent over the network from a file. I am not sure if that would be enough.

**Comment by dwalluck** [comment]
I am merging this because the current PR no longer matches its original purpose

**Pull request merged**